packages/app/src/clickhouse.ts

```diff
@@ -216,6 +216,23 @@ export class ClickHouseQueryError extends Error {
   }
 }
 
+export function extractColumnReference(
+  sql: string,
+  maxIterations = 10,
+): string | null {
+  let iterations = 0;
+
+  // Loop until we remove all function calls and get just the column, with a maximum limit
+  while (/\w+\([^()]*\)/.test(sql) && iterations < maxIterations) {
+    // Replace the outermost function with its content
+    sql = sql.replace(/\w+\(([^()]*)\)/, '$1');
+    iterations++;
+  }
+
+  // If we reached the max iterations without resolving, return null to indicate an issue
+  return iterations < maxIterations ? sql.trim() : null;
```

Comment:
what if it resolves at the last iteration? do we get null? usually I'd add a few simple unit tests for function like this. on the other hand, I wonder if we want to use sql parser here

Reply:
I'm not sure this is a material consideration for the PR (this is also not a new function, I'm just refactoring it here)

Reply:
I realized it was just code refactoring but still wanted to put this on the radar. we can revisit it later given its a rare edge case